### ISSUE TITLE
Pivot table visualisation

### ISSUE THREAD

**Issue by jgough** [opensearch-project/OpenSearch-Dashboards#705]
A long-standing feature request on Kibana since 2015 is to have a [pivot table visualisation](https://github.com/elastic/kibana/issues/5049) of some sort.

At the time of writing this issue has 50 thumbs-up reactions making it one of the most requested features in Kibana that has not been implemented.

I don't know exactly how this would work from a UI perspective but being able to split a series and add terms as additional columns would open up a lot of possibilities for making much richer (and easier to understand) data tables.

There are a couple of implementations that others seem to have already done that provide pivoting capabilities:
https://www.flexmonster.com/blog/new-pivot-table-for-kibana/
https://github.com/fbaligand/kibana-enhanced-table

Bringing this feature request over to here since there seems to be high demand for this.

**Comment by JacobBrandt** [comment]
An implementation of a pivot table goes hand in hand with a pivot chart which is one of the features of what I envision #379 should provide.

**Comment by joshuarrrr** [comment]
Yeah, I think this is a no-brainer in terms of useful functionality to add. We may be able to leverage our recent refactoring of the table visualization to tackle this sooner than later.

**Comment by ashwin-pc** [comment]
@joshuarrrr +1. I took a deeper look at this today and i think we can repurpose the table vis in VisBuilder to behave as a pivot table instead. It already has the other features needed to make this work. For users who want a more specific table viz they can always resort to the original table viz type while pivot table users can simply rely on VisBuilder. Also there is no real difference between a basic table vis and a pivot table.
